### PR TITLE
Add unistd::{access, faccessat}

### PR DESCRIPTION
**Pull request by bugaevc**
We should also add `AT_EACCESS` (and the rest of `AT_*` flags), but that's not included in this PR.

### CODE REVIEW

File: src/unistd.rs [hunk 1]
```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
...
 
+libc_bitflags!{
+    pub flags AccessFlags: c_int {
+        R_OK,
+        W_OK,
+        X_OK,
+    }
+}
+
+pub enum AccessMode {
```
Comment (Susurrus) [inline src/unistd.rs]: This should be documented as it's a public interface.

File: src/unistd.rs [hunk 2]
```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
     Ok((fd, PathBuf::from(pathname)))
 }
 
+libc_bitflags!{
+    pub flags AccessFlags: c_int {
```
Comment (Susurrus) [inline src/unistd.rs]: Please document this.

File: src/unistd.rs [hunk 3]
```diff
@@ -1,14 +1,14 @@
 //! Safe wrappers around functions found in libc "unistd.h" header
 
 use {Errno, Error, Result, NixPath};
-use fcntl::{fcntl, OFlag, O_CLOEXEC, FD_CLOEXEC};
+use fcntl::{fcntl, OFlag, AtFlags, O_CLOEXEC, FD_CLOEXEC};
```
Comment (Susurrus) [inline src/unistd.rs]: This should be alphabetical, so `AtFlags` is first.

File: src/unistd.rs [hunk 4]
```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
...
+        X_OK,
+    }
+}
+
+pub enum AccessMode {
+    F_OK,
+    Flags(AccessFlags),
+}
+
+impl AccessMode {
```
Comment (Susurrus) [inline src/unistd.rs]: This should be commented.

File: src/unistd.rs [hunk 5]
```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
...
+    fn to_libc_type(&self) -> c_int {
+        match self {
+            &AccessMode::F_OK => libc::F_OK,
+            &AccessMode::Flags(flags) => flags.bits(),
+        }
+    }
+}
+
+/// Check whether a file exists or whether the current process can access a file.
```
Comment (Susurrus) [inline src/unistd.rs]: Please link to the man page for this. See the `fork()` function doccomments for an example.

File: src/unistd.rs [hunk 6]
```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
...
+pub fn access<P: ?Sized + NixPath>(path: &P, mode: AccessMode) -> Result<()> {
+    let res = try!(path.with_nix_path(|path|
+        unsafe {
+            libc::access(path.as_ptr(), mode.to_libc_type())
+        }
+    ));
+    Errno::result(res).map(drop)
+}
+
+/// Check whether a file exists or whether the current process can access a file,
```
Comment (Susurrus) [inline src/unistd.rs]: Please link to the man page for this. See the `fork()` function doccomments for an example.

**Comment by Susurrus** [comment]
Thanks for your PR, @bugaevc! I imagine this is pretty useful. I've made some comments, mostly superficial stuff right now. I'll come back and make a deeper inspection of stuff once things are a little more comments. Also, when you make these changes, could you squash everything into 1 commit? Thanks!

**Comment by bugaevc** [comment]
☑️  Done

**Comment by Susurrus** [comment]
I just realized that this PR overlaps with #562, which covers a lot more ground. I'd like to close this PR in favor of that one. Would you be willing to help me review that PR and we can close this one?

**Comment by bugaevc** [comment]
Sure -- I wonder how come I haven't noticed it myself.

**Pull request closed without merging**